base/common/src/main/java/com/netscape/certsrv/account/Account.java

```diff
@@ -133,51 +135,127 @@ public boolean equals(Object obj) {
     }
 
     public String toXML() throws Exception {
-        Marshaller marshaller = JAXBContext.newInstance(Account.class).createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
 
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.newDocument();
+
+        Element rootElement = document.createElement("Account");
+        rootElement.setAttribute("id", id);
+        document.appendChild(rootElement);
+
+        // The original XML mapping always creates Attributes element even if it's empty.
+        Element attrsElement = document.createElement("Attributes");
+        rootElement.appendChild(attrsElement);
+
+        for (Map.Entry<String, String> attribute : attributes.entrySet()) {
+            Element attrElement = document.createElement("Attribute");
+            attrElement.setAttribute("name", attribute.getKey());
+            attrElement.appendChild(document.createTextNode(attribute.getValue()));
+            attrsElement.appendChild(attrElement);
+        }
+
+        if (className != null) {
+            Element classNameElement = document.createElement("ClassName");
+            classNameElement.appendChild(document.createTextNode(className));
+            rootElement.appendChild(classNameElement);
+        }
+
+        if (fullName != null) {
+            Element fullNameElement = document.createElement("FullName");
+            fullNameElement.appendChild(document.createTextNode(fullName));
+            rootElement.appendChild(fullNameElement);
+        }
+
+        if (email != null) {
+            Element emailElement = document.createElement("Email");
+            emailElement.appendChild(document.createTextNode(email));
+            rootElement.appendChild(emailElement);
+        }
+
+        if (!roles.isEmpty()) {
+            Element rolesElement = document.createElement("Roles");
+            rootElement.appendChild(rolesElement);
+
+            for (String role : roles) {
+                Element roleElement = document.createElement("Role");
+                roleElement.appendChild(document.createTextNode(role));
+                rolesElement.appendChild(roleElement);
```

Comment:
Minor: Is it worth pulling these few lines into a helper method we do 'create, add text, add to root' a few times

Could look something like:
```java 

private void createAndAddElement(Element rootElement, String elementName, String elementValue) {
    Element element = document.createElement(elementName);
    emailElement.appendChild(document.createTextNode(elementValue));
    rootElement.appendChild(element);
}

```

Reply:
It's possible, but we need to figure out where to put this method since we're going to need it in a number of classes. I think we should see some more classes with similar code before we can decide.

Reply:
After working on several more classes I noticed some variations in these codes due to type conversions. Since this is just a temporary code I'll just keep copying the code for simplicity. Feel free to drop the new `toDOM()`, `fromDOM()`, `toXML()`, and `fromXML()` if it doesn't break CI.